docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/configuration-editor.md

```diff
@@ -0,0 +1,91 @@
+---
+id: configuration-editor
+title: The configuration editor
+description: Testing the configuration editor
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - testing
+  - e2e
+  - data-source
+  - configuration editor
+sidebar_position: 10
+---
+
+## Introduction
+
+Most data source plugins need auth to communicate with third-party services. Users can configure the connection details in the data source configuration page. Data source plugins that need auth cannot function without valid configuration, so it's crucial that the configuration page and the data source health check endpoint that is used to test the configuration work as expected.
+
+### Testing the configuration in a backend data source plugin
+
+In the following example, the configuration of a backend data source plugin is tested. The configuration editor form is populated with valid configuration values, and then the `Save & test` button is clicked. Clicking `Save & test` calls Grafana backend to save the configuration and then passes configuration to the plugin's backend health check endpoint. The test will be successful only if both calls yields a successful status code.
+
+```ts title="configurationEditor.spec.ts"
+test('"Save & test" should be successful when configuration is valid', async ({
+  createDataSourceConfigPage,
+  readProvisionedDataSource,
+  page,
+}) => {
+  const ds = readProvisionedDataSource<JsonData, SecureJsonData>({ fileName: 'datasources.yaml' });
+  const configPage = await createDataSourceConfigPage({ type: ds.type });
+  await page.getByLabel('Path').fill(ds.jsonData.path);
+  await page.getByLabel('API Key').fill(ds.secureJsonData.apiKey);
+  await expect(configPage.saveAndTest()).toBeOK();
+});
+```
+
+### When configuration is invalid
```

Comment:
if this is backend specific then should be h4, if not, might be better after the frontend section?

Reply:
It's backend specific, so changing to h4 👍